concent_api/core/views.py

```diff
@@ -1241,6 +1209,712 @@ def store_message_and_message_status(
+    client_public_key:  bytes   = None,
+    message_inbox_type: Enum    = None,
+):
+    assert client_public_key    not in ['', None]
+    assert message_inbox_type   in PendingResponse.PendingResponseType
+    database_message = PendingResponse.objects.filter(
+        client__public_key = b64encode(client_public_key),
+        queue              = message_inbox_type.name,
+        delivered          = False,
+    ).order_by('created_at').first()
+
+    if database_message is None:
+        return None
+
+    if database_message.response_type == PendingResponse.ResponseType.ForceReportComputedTask.name:  # pylint: disable=no-member
+        report_computed_task = deserialize_message(database_message.subtask.report_computed_task.data.tobytes())
+        mark_message_as_delivered(database_message)
+        response_to_client = message.concents.ForceReportComputedTask(
+            report_computed_task = report_computed_task
+        )
+        logging.log_receive_message_from_database(
+            response_to_client,
+            database_message.client.public_key,
+            database_message.response_type,
+            database_message.queue
+            )
+        return response_to_client
+
+    elif database_message.response_type == PendingResponse.ResponseType.AckReportComputedTask.name:  # pylint: disable=no-member
+        if database_message.subtask.ack_report_computed_task is not None:
+            ack_report_computed_task = deserialize_message(database_message.subtask.ack_report_computed_task.data.tobytes())
+            mark_message_as_delivered(database_message)
+            response_to_client = message.concents.ForceReportComputedTaskResponse(
+                ack_report_computed_task = ack_report_computed_task,
+            )
+            logging.log_receive_message_from_database(
+                response_to_client,
+                database_message.client.public_key,
+                database_message.response_type,
+                database_message.queue
+            )
+            return response_to_client
+        else:
+            ack_report_computed_task = message.concents.AckReportComputedTask(
+                task_to_compute = deserialize_message(database_message.subtask.task_to_compute.data.tobytes())
+            )
+            add_messages_to_subtask(
+                database_message.subtask,
+                ack_report_computed_task = ack_report_computed_task,
+            )
+            database_message.subtask.full_clean()
+            database_message.subtask.save()
+            mark_message_as_delivered(database_message)
+            response_to_client = message.concents.ForceReportComputedTaskResponse(
+                ack_report_computed_task = ack_report_computed_task
+            )
+            logging.log_receive_message_from_database(
+                response_to_client,
+                database_message.client.public_key,
+                database_message.response_type,
```

Comment:
This code for logging and marking the message as delivered repeats everywhere. Maybe it would be better to have the `if` only create the message and do the logging and marking after the `if`?